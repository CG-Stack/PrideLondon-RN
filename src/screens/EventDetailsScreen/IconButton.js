// @flow
import React from "react";
import { Image, StyleSheet } from "react-native";
import Touchable from "../../components/Touchable";

type Props = {
  onPress: Function,
  source: Image.propTypes.source
};

const IconButton = ({ onPress, source }: Props) => (
  <Touchable onPress={onPress}>
    <Image style={styles.touchableIcon} source={source} />
  </Touchable>
);

const styles = StyleSheet.create({
  touchableIcon: {
    width: 48,
    height: 48
  }
});

export default IconButton;
