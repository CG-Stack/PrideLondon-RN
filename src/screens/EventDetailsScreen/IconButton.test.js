// @flow
import React from "react";
import { shallow } from "enzyme";
import IconButton from "./IconButton";

it("renders correctly", () => {
  const output = shallow(
    <IconButton
      onPress={() => {}}
      source={{ uri: "/super-awesome-image.jpg" }}
    />
  );
  expect(output).toMatchSnapshot();
});
