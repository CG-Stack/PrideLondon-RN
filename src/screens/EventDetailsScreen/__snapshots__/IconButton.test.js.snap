// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#20215a",
    }
  }
>
  <withOrientation>
    <StatusBar
      animated={true}
      barStyle="light-content"
      showHideTransition="fade"
    />
  </withOrientation>
</View>
`;
